test/unit/test_matrix.cpp

```diff
@@ -1153,3 +1153,47 @@ TEST(MatrixDestructorFutures, ConstAfterRead) {
 
   last_task.get();
 }
+
+struct CustomException final : public std::exception {};
+
+TEST(MatrixExceptionPropagation, PropagateOnRwWithRWAccess) {
+  auto matrix = createMatrix<TypeParam>();
+
+  auto f =
+      matrix(LocalTileIndex(0, 0)).then(hpx::util::unwrapping([](auto&&) { throw CustomException{}; }));
+
+  EXPECT_THROW(matrix(LocalTileIndex(0, 0)).get(), dlaf::ContinuationException);
+  EXPECT_THROW(f.get(), CustomException);
+}
+
+TEST(MatrixExceptionPropagation, PropagateOnRwWithReadAccess) {
+  auto matrix = createMatrix<TypeParam>();
+
+  auto f =
+      matrix(LocalTileIndex(0, 0)).then(hpx::util::unwrapping([](auto&&) { throw CustomException{}; }));
+
+  EXPECT_THROW(matrix.read(LocalTileIndex(0, 0)).get(), dlaf::ContinuationException);
+  EXPECT_THROW(f.get(), CustomException);
+}
+
+TEST(MatrixExceptionPropagation, NoPropagateOnReadOnlyWithRWAccess) {
+  auto matrix = createMatrix<TypeParam>();
+
+  auto f = matrix.read(LocalTileIndex(0, 0)).then(hpx::util::unwrapping([](auto&&) {
+    throw CustomException{};
+  }));
+
+  EXPECT_NO_THROW(matrix(LocalTileIndex(0, 0)).get());
+  EXPECT_THROW(f.get(), CustomException);
+}
+
+TEST(MatrixExceptionPropagation, NoPropagateOnReadOnlyWithReadAccess) {
+  auto matrix = createMatrix<TypeParam>();
+
+  auto f = matrix.read(LocalTileIndex(0, 0)).then(hpx::util::unwrapping([](auto&&) {
+    throw CustomException{};
+  }));
+
+  EXPECT_NO_THROW(matrix.read(LocalTileIndex(0, 0)).get());
+  EXPECT_THROW(f.get(), CustomException);
+}
```

Comment:
Please be consistent with test naming:
Rw <-> RW
ReadOnly <-> Read
What about e.g. `RWPropagatesInReadAccess`?